File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> largestValues(TreeNode* root) {
        vector<int> ans;
        function<void(TreeNode*, int)> levelOrder = [&](TreeNode* root, int level) {
            if(!root) return;
            if(level >= ans.size()) ans.push_back(root->val);
            else if(root->val > ans[level]) ans[level] = root->val;
            levelOrder(root->left, level + 1);
            levelOrder(root->right, level + 1);
        };
        levelOrder(root, 0);
        return ans;
    }
};

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        ans = []
        def levelOrder(root, level):
            if root == None: 
                return
            if level >= len(ans):
                ans.append(root.val)
            elif root.val > ans[level]:
                ans[level] = root.val
            levelOrder(root.left, level + 1)
            levelOrder(root.right, level + 1)
        levelOrder(root, 0)
        return ans